### PR TITLE
A Tiny JS World task

### PR DESCRIPTION
**Pull request by franchukv**
### A Tiny JS World

[Demo](https://franchukv.github.io/a-tiny-JS-world/) | [Code base](https://github.com/franchukv/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by franchukv** [comment]
Self-check done.

File: submissions/franchukv/a-tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+const objToString = (obj) => {
+   const propNames = Object.getOwnPropertyNames(obj);
+   let propValuesArr = [];
+
+   for (let i = 0; i < propNames.length; i++) {
+      propValuesArr.push(obj[propNames[i]]);
+   }
+
+   return propValuesArr.join('; ');
```
Comment (OleksiyRudenko) [inline submissions/franchukv/a-tiny-js-world/index.js:79]: CLassic use case for `Array#map`. You won't also need interim variables.

File: submissions/franchukv/a-tiny-js-world/index.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+   saying: cat.saying,
+   friends: `In astral connection with ${cat.name}.`,
+};
+
+// ======== OUTPUT ========
+
+const inhabitants = [man, woman, dog, cat, catwoman];
+
+const objToString = (obj) => {
+   const propNames = Object.getOwnPropertyNames(obj);
```
Comment (OleksiyRudenko) [inline submissions/franchukv/a-tiny-js-world/index.js:72]: > - [ ] `Object` methods like `keys`, `values`, `entries` shouldn't be used when a particular order is required as these do not guarantee any particular order of keys/values. Same refers to `for...of` and `for...in` when applied to objects.

The key message here ^^ is **as these do not guarantee any particular order of keys/values**. Same refers to any other `Object` methods. Please use approach when order is controllable and manageable.

File: submissions/franchukv/a-tiny-js-world/index.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+   feets: 2,
+   saying: cat.saying,
+   friends: `In astral connection with ${cat.name}.`,
+};
+
+// ======== OUTPUT ========
+
+const inhabitants = [man, woman, dog, cat, catwoman];
+
+const objToString = (obj) => {
```
Comment (OleksiyRudenko) [inline submissions/franchukv/a-tiny-js-world/index.js:71]: > - Give variables and functions meaningful names. Avoid generic names like `item`, `element`, `key`, `object`, `array` or their variations. Exception: helper functions that are specifically and intentionally designed to be multipurpose.

File: submissions/franchukv/a-tiny-js-world/index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+   let propValuesArr = [];
+
+   for (let i = 0; i < propNames.length; i++) {
+      propValuesArr.push(obj[propNames[i]]);
+   }
+
+   return propValuesArr.join('; ');
+};
+
+function greetings(arr, func) {
```
Comment (OleksiyRudenko) [inline submissions/franchukv/a-tiny-js-world/index.js:82]: > - Give variables and functions meaningful names. Avoid generic names like `item`, `element`, `key`, `object`, `array` or their variations. Exception: helper functions that are specifically and intentionally designed to be multipurpose.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@franchukv good job!
Also a few of improvements are needed.

File: submissions/franchukv/a-tiny-js-world/index.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+const properties = ['species', 'gender', 'name', 'status', 'hands', 'feets', 'saying', 'friends'];
+
+function objectToString(habitant) {
+   let temporaryArray = [];
+
+   properties.forEach((property) => {
+      temporaryArray.push(habitant[property]);
+   })
+
+   return temporaryArray.join('; ');
```
Comment (OleksiyRudenko) [inline submissions/franchukv/a-tiny-js-world/index.js:74]: rows 68-74 are classic use case for `Array#map`. Learn it. You will often use it.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@franchukv still need to master `Array#map`

**Review by OleksiyRudenko — APPROVED** [review]
@franchukv well done!

**Comment by franchukv** [comment]
> @franchukv well done!

Thanks for ur review)

**Pull request merged**